### PR TITLE
Remove dead code

### PR DESCRIPTION
**Pull request by sdmaclea**

### CODE REVIEW

File: src/vm/appdomain.hpp [hunk 1]
```diff
@@ -2603,48 +2603,13 @@ class AppDomain : public BaseDomain
...
     }
 
-    void SetPassiveDomain()
-    {
-        LIMITED_METHOD_CONTRACT;
-
-        m_dwFlags |= PASSIVE_DOMAIN;
-    }
-
     BOOL IsPassiveDomain()
```
Comment (jkotas) [inline src/vm/appdomain.hpp]: If nobody sets it and it is always false, we can delete it too (and fix the few callers that call it).

Comment (sdmaclea) [reply src/vm/appdomain.hpp]: The flag is set by `SetCompilationDomain()`  which I think is used in crossgen.

I can change all the `IsPassiveDomain()` to `IsCompilationDomain()`

Comment (jkotas) [reply src/vm/appdomain.hpp]: Ok, I have missed that. You can leave it as it is, or change all `IsPassiveDomain()` to `IsCompilationDomain()`. Either way is fine with me.

**Comment by sdmaclea** [comment]
I wanted to get rid of IsUserDomain() but that flag is set if the friendly name for the domain is set.  Can we get rid of the friendly name and always use the `DEFAULT_DOMAIN_FRIENDLY_NAME` string.

**Comment by sdmaclea** [comment]
Ignoring arm & arm64 failures.  These are failing on all CI runs with errors about 'TypeEquivalenceTypes'

**Comment by jkotas** [comment]
> I wanted to get rid of `IsUserDomain()`

We should be assuming that all domains are user domains. I actually got rid of that flag already in the cleanup delta that I am working on right now (getting rid of the unmanaged->managed->unmanaged roundtrip for all the config vars).

**Pull request merged**